config/environments/test.rb

```diff
@@ -51,10 +51,10 @@
   # config.action_view.annotate_rendered_view_with_filenames = true
 
   # Devise mailer.
-  config.action_mailer.default_url_options = {host: "localhost", port: 3000}
+  config.action_mailer.default_url_options = {host: "localhost", port: 3000, locale: I18n.locale}
 
   # Configure host for URL helpers.
-  Rails.application.routes.default_url_options = {host: "localhost", port: 3000}
+  Rails.application.routes.default_url_options = {host: "localhost", port: 3000, locale: I18n.locale}
```

Comment:
Why are these only added for test but not for any other environment? And what is the purpose of them?

When in a configuration file `I18n.locale` will always be English, so we should be able to omit this, right?

Reply:
This is not needed after set `locale nil` in `routes.rb`.

Reply:
You are right! I also need to set this for development environment to make notifications & mailers work.
[Here](https://github.com/excid3/noticed/issues/49) is a reference.
And instead of `I18n.locale`, I use `{locale: nil}`. Because currently there is no mechanism to set locale param to nil when it's the default locale.